Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -948,8 +948,16 @@ void Document::update_style()
         return;
 
     evaluate_media_rules();
-    if (update_style_recursively(*this))
+
+    auto invalidation = update_style_recursively(*this);
+    if (invalidation.rebuild_layout_tree) {
         invalidate_layout();
+    } else {
+        if (invalidation.relayout)
+            set_needs_layout();
+        if (invalidation.rebuild_stacking_context_tree)
+            invalidate_stacking_context_tree();
+    }
     m_needs_full_style_update = false;
```

Comment:
I think this line shouldn't be here, since this line will be executed if the invalidation requires relayout or stacking context tree changed.

I haven't looked into the effects of this boolean, but I'm guessing some style invalidation is currently broken.

Reply:
At this point, we have just completed a style update. Hence, we no longer need to do a style update, full or partial :)